### PR TITLE
refactor: make patch state key `Vec<u8>` instead of `String`

### LINKED ISSUE #51: patch state key should be `Vec<u8>` and not `String`
Currently only usage uses string keys, but this should not be required to be utf8 bytes

### PR DESCRIPTION
**Pull request by itegulov**
Fixes #51

### CODE REVIEW

**Review by ChaoticTempest — APPROVED** [review]

File: workspaces/src/network/mod.rs [hunk 1]
```diff
@@ -120,12 +120,12 @@ where
     async fn patch_state(
         &self,
         contract_id: &AccountId,
-        key: String,
+        key: Vec<u8>,
         value: Vec<u8>,
```
Comment (austinabell) [inline workspaces/src/network/mod.rs:124]: these should probably be slices to be consistent with the rest of the API, or is there a reason they are not @ChaoticTempest ?

Comment (ChaoticTempest) [reply workspaces/src/network/mod.rs:124]: ahh, yeah best to keep them as slices. My bad, prematurely merged this one since it looked simple

Comment (itegulov) [reply workspaces/src/network/mod.rs:124]: no worries, I can fix this real quick, I just went by the issue description

you want me to change both key and value to be [u8], right?

Comment (ChaoticTempest) [reply workspaces/src/network/mod.rs:124]: yup both should be slices

Comment (austinabell) [reply workspaces/src/network/mod.rs:124]: was unrelated to this PR, just wanted to point out before opening an issue :)

**Pull request merged**